packages/blade/src/components/Dropdown/Dropdown.tsx

```diff
@@ -74,11 +76,26 @@ const _Dropdown = ({
   const dropdownBaseId = useId('dropdown');
 
   const dropdownTriggerer = React.useRef<DropdownContextType['dropdownTriggerer']>();
+  const isFirstRenderRef = React.useRef(true);
+
+  React.useEffect(() => {
+    // Ignoring the `onDismiss` call on first render
+    if (isFirstRenderRef.current) {
+      isFirstRenderRef.current = false;
+      return;
+    }
+
+    if (!isOpen && onDismiss) {
+      onDismiss();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOpen]);
```

Comment:
Instead of relying on isOpen state and syncing it to call the onDismiss. 

Won't it be better to actually call the onDismiss in places where we did setIsOpen(false). 
That would fix the firstRender ignore edge case too.

Reply:
yup wanted to do that only. Except to do that, we'll have to set the `onDismiss` callback in the DropdownContext. Wanted to avoid that because we already have so many things there.

Reply:
You don't need to pass onDismiss into the DropdownContext. 

You could: 

```
const close = useCallback(() => {
  setIsOpen(false);
  onDismiss?.();
})
```

and just pass the close function downstream and whenever you setIsOpen(false) call `close`

Reply:
> pass the close function downstream

How exactly 🤔 Because Dropdown is a parent component and we need to call close inside `ActionList`, `DropdownOverlay`, and from `useDropdown`. So we will have to set it in context right?

Reply:
oh wait I think we will have to set it in context but it will work with useCallback. Lemme try.

Reply:
done. should look better now.